sdk/storage/storage-blob/execute-samples.js

```diff
@@ -0,0 +1,76 @@
+const execa = require("execa");
+const fs = require("fs");
+require("dotenv").config({ path: "../.env" });
+
+// Samples can be skipped by mentioning them in the skipSet
+const skipSet = ["proxyAuth"];
+
+const bDel = `!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!`;
+const del = `${bDel}------------------------------${bDel}`;
+
+const { g, y, r, b } = [["r", 1], ["g", 2], ["b", 4], ["y", 3]].reduce(
+  (cols, col) => ({
+    ...cols,
+    [col[0]]: (f) => `\x1b[3${col[1]}m${f}\x1b[0m`
+  }),
+  {}
+);
+
+async function exec(cmd, cwd) {
+  let command = execa(cmd, {
+    cwd,
+    shell: true
+  });
+  command.stderr.pipe(process.stderr);
+  command.stdout.pipe(process.stdout);
+  return command;
+}
+
+async function runSamples(language) {
+  let cmd;
+  const directory = `./samples/${language}`;
+
+  if (language === "typescript") {
+    cmd = "ts-node";
+  } else {
+    cmd = "node";
+    await exec(`npm run build:js-samples`, directory);
+  }
+
+  console.log(`Running ${language} samples...`);
+
+  const files = fs.readdirSync(directory);
```

Comment:
nit: is there an async version for `fs.readdir`?

Reply:
yes, there is

Reply:
(Async)`fs.readdir()` requires a callback which would force us to do everything inside the callback.

or

We can Promisify `fs.readdir()` and then use it.
```javascript
fs.readdirAsync = function(dirname) {
  return new Promise(function(resolve, reject) {
    fs.readdir(dirname, function(err, filenames) {
      if (err) reject(err);
      else resolve(filenames);
    });
  });
};
```

Not doing this since `fs.readdirSync` is enough for our limited number of samples.